Fastly/fastly-waf/ingest/parser.yml

```
@@ -0,0 +1,50 @@
+name: fastly-waf
+pipeline:
+  - name: parse_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: "{{original.message}}"
+        output_field: message
+
+  - name: set_fields
+
+stages:
+  set_fields:
+    actions:
+      - set:
+          "@timestamp": "{{parse_event.message.timestamp}}"
+
+          observer.vendor: "Fastly"
+          observer.product: "Fastly Next-Gen WAF"
+
+          event.kind: "event"
+          event.module: "fastly.waf"
+          event.category: "network"
+
+      - translate:
+        dictionary:
+          "flagged": ["denied"]
+          "info": ["info"]
+        fallback: ["info"]
+        mapping:
+          parse_event.message.action: event.type
+
+      - set:
+          user_agent.original: "{{parse_event.message.userAgents[0]}}"
+        filter: "{{parse_event.message.userAgents != null }}"
+
+      - set:
+          action.name: "{{parse_event.message.action}}"
+          source.ip: "{{parse_event.message.source}}"
+
+          host.geo.country_iso_code: "{{parse_event.message.remoteCountryCode}}"
+          host.hostname: "{{parse_event.message.remoteHostname}}"
```

I suggest using `host.name` instead of [`host.hostname`](https://www.elastic.co/guide/en/ecs/8.11/ecs-host.html#field-host-hostname) (that is too restrictive in its definition)
```suggestion
          host.name: "{{parse_event.message.remoteHostname}}"
```